pkg/controller/dnszone/azureactuator.go

```diff
@@ -114,26 +114,26 @@ func (a *AzureActuator) ModifyStatus() error {
 	}
 
 	a.dnsZone.Status.Azure = &hivev1.AzureDNSZoneStatus{
-		ZoneName: a.managedZone.Name,
+		ZoneID: a.managedZone.ID,
 	}
 
 	return nil
 }
 
 // Refresh implements the Refresh call of the actuator interface
 func (a *AzureActuator) Refresh() error {
-	var zoneName string
-	if a.dnsZone.Status.Azure != nil && a.dnsZone.Status.Azure.ZoneName != nil {
-		a.logger.Debug("ZoneName is set in status, will retrieve by that name")
-		zoneName = *a.dnsZone.Status.Azure.ZoneName
+	if a.managedZone == nil {
+		return nil
 	}
 
-	resourceGroupName := a.dnsZone.Spec.Azure.BaseDomainResourceGroupName
+	zoneName := a.managedZone.Name
```

Comment:
Do not use the name from `managedZone`. Use the name from the spec of the `DNSZone`. The `managedZone` is what this function is refreshing.

Reply:
Fixed